File: app/local_plugins/agent_chat/android/src/main/kotlin/com/aiagent/agent_chat/ChatAgentFlutterPlugin.kt
package com.aiagent.agent_chat

import android.util.Log
import com.aiagent.plugin_interface.NativeAgentRegistry
import io.flutter.embedding.engine.plugins.FlutterPlugin

/**
 * ChatAgentFlutterPlugin — Chat Agent Flutter 插件入口
 *
 * 在 onAttachedToEngine 时注册 "chat" Agent 类型到 NativeAgentRegistry。
 * agents_server 通过 NativeAgentRegistry.create("chat") 创建 ChatAgentSession 实例。
 */
class ChatAgentFlutterPlugin : FlutterPlugin {

    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        NativeAgentRegistry.register("chat") { ChatAgentSession() }
        Log.d("ChatAgentPlugin", "Registered NativeAgent type=chat")
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {}
}

File: app/local_plugins/agents/agent_chat/ios/Classes/ChatAgentFlutterPlugin.swift
import Flutter
import ai_plugin_interface
import os.log

/// Chat agent Flutter plugin entry.
///
/// Registers a factory for the `"chat"` agent type with the global
/// `NativeAgentRegistry`. `agents_server` instantiates an agent via
/// `NativeAgentRegistry.shared.create("chat")` when the Dart side calls
/// `createAgent(agentType:"chat", …)`.
public class ChatAgentFlutterPlugin: NSObject, FlutterPlugin {
    private static let log = OSLog(subsystem: "com.aiagent.agent_chat", category: "Plugin")

    public static func register(with registrar: FlutterPluginRegistrar) {
        NativeAgentRegistry.shared.register("chat") { ChatAgentSession() }
        os_log("Registered NativeAgent type=chat", log: log, type: .debug)
    }
}
